### PR TITLE
[Work in progress] 1.2 feature set

### PR DESCRIPTION
**Pull request by dclaux**

### CODE REVIEW

**Comment by dclaux** [comment]
@paulcam206 this is now fixed.

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 1]
```diff
@@ -3115,8 +3223,62 @@ export class OpenUrlAction extends Action {
...
+		for (let elementId of Object.keys(this.targetElements)) {
+			let targetElement = this.parent.getRootElement().getElementById(elementId);
+
+			if (targetElement) {
+				if (typeof this.targetElements[elementId] === "boolean") {
+					targetElement.isVisible = this.targetElements[elementId];
+				} 
+				else {
+                    targetElement.isVisible = !targetElement.isVisible;
+				}
```
Comment (paulcam206) [inline source/nodejs/adaptivecards/src/card-elements.ts]: looks like we're mixing tabs and spaces

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 2]
```diff
@@ -3882,263 +4058,354 @@ export class ActionSet extends CardElement {
...
-					return false;
+				this.renderedElement.style.marginLeft = "-" + parentPhysicalPadding.left + "px";
+				this.renderedElement.style.marginRight = "-" + parentPhysicalPadding.right + "px";
+
+				if (this.separatorElement && this.separatorOrientation == Enums.Orientation.Horizontal) {
+					this.separatorElement.style.marginLeft = "-" + parentPhysicalPadding.left + "px";
+					this.separatorElement.style.marginRight = "-" + parentPhysicalPadding.right + "px";
 				}
 			}
+			*/
```
Comment (paulcam206) [inline source/nodejs/adaptivecards/src/card-elements.ts]: can probably just get rid of this

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 3]
```diff
@@ -4821,40 +4994,39 @@ export class ColumnSet extends CardElementContainer {
...
 					// Best way to emulate "internal" access I know of
 					column["_computedWeight"] = computedWeight;
 				}
 
-				var renderedColumn = column.render();
+				let renderedColumn = column.render();
 
 				if (renderedColumn) {
-					if (renderedColumnCount > 0 && column.separatorElement) {
+					// if (renderedColumnCount > 0 && column.separatorElement) {
```
Comment (paulcam206) [inline source/nodejs/adaptivecards/src/card-elements.ts]: >// if (renderedColumnCount > 0 && column.separatorElement) { [](start = 5, length = 60)

unneeded comment

**Pull request merged**